bedboss/bedstat/bedstat.py

```diff
@@ -0,0 +1,229 @@
+from hashlib import md5
+from typing import NoReturn
+import json
+import yaml
+import os
+import requests
+import gzip
+import pypiper
+import bbconf
+
+
+SCHEMA_PATH_BEDSTAT = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "pep_schema.yaml"
+)
+
+
+def hash_bedfile(filepath: str) -> str:
+    """
+    Generate digest for bedfile
+    :param str filepath: path to the bed file
+    :return str: digest of the files
+    """
+    with gzip.open(filepath, "rb") as f:
+        # concate column values
+        chrs = ",".join([row.split()[0].decode("utf-8") for row in f])
+        starts = ",".join([row.split()[1].decode("utf-8") for row in f])
+        ends = ",".join([row.split()[2].decode("utf-8") for row in f])
+        # hash column values
+        chr_digest = md5(chrs.encode("utf-8")).hexdigest()
+        start_digest = md5(starts.encode("utf-8")).hexdigest()
+        end_digest = md5(ends.encode("utf-8")).hexdigest()
+        # hash column digests
+        bed_digest = md5(
+            ",".join([chr_digest, start_digest, end_digest]).encode("utf-8")
+        ).hexdigest()
+
+        return bed_digest
+
+
+def convert_unit(size_in_bytes: int) -> str:
+    """
+    Convert the size from bytes to other units like KB, MB or GB
+    :param int size_in_bytes: size in bytes
+    :return str: File size as string in different units
+    """
+    if size_in_bytes < 1024:
+        return str(size_in_bytes) + "bytes"
+    elif size_in_bytes in range(1024, 1024 * 1024):
+        return str(round(size_in_bytes / 1024, 2)) + "KB"
+    elif size_in_bytes in range(1024 * 1024, 1024 * 1024 * 1024):
+        return str(round(size_in_bytes / (1024 * 1024))) + "MB"
+    elif size_in_bytes >= 1024 * 1024 * 1024:
+        return str(round(size_in_bytes / (1024 * 1024 * 1024))) + "GB"
+
+
+def bedstat(
+    bedfile: str,
+    bedbase_config: str,
+    genome: str,
+    outfolder: str,
+    ensdb: str = None,
+    open_signal_matrix: str = None,
+    bigbed: str = None,
+    sample_yaml: str = None,
+    just_db_commit: bool = False,
+    no_db_commit: bool = False,
+    force_overwrite: bool = False,
+    pm: pypiper.PipelineManager = None,
+    **kwargs,
+) -> NoReturn:
+    """
+    Run bedstat pipeline. Can be used without running from command line
```

Comment:
Can you explain, "Can be used without running from command line" ? Not sure this is a helpful docstring

Reply:
It would be better to describe what this function does

Reply:
A major part of it appears to be inserting something into a database